app/src/main/java/io/zenandroid/onlinego/ui/screens/stats/StatsFragment.kt

```diff
@@ -34,5 +73,301 @@ class StatsFragment : Fragment(), StatsContract.View {
         presenter.unsubscribe()
     }
 
+    override fun fillPlayerDetails(playerDetails: OGSPlayer ) {
+        nameView.text = playerDetails.username
+        rankView.text = formatRank(egfToRank(playerDetails.ratings?.overall?.rating))
+        playerDetails.country?.let {
+            flagView.text = convertCountryCodeToEmojiFlag(it)
+        }
+        playerDetails.icon?.let {
+            Glide.with(this)
+                    .load(processGravatarURL(it, iconView.width))
+                    .transition(DrawableTransitionOptions.withCrossFade(DrawableCrossFadeFactory.Builder().setCrossFadeEnabled(true).build()))
+                    .apply(RequestOptions().centerCrop().placeholder(R.drawable.ic_person_outline))
+                    .into(iconView)
+        }
+
+        var rankViewText = SpannableString("${formatRank(egfToRank(playerDetails.ratings?.overall?.rating))}\n" +
```

Comment:
Wouldn't it be simpler to have 2 text views here with different styles in the XML? This makes the code very hard to read...

Reply:
I agree. I did add grids of layoutcontainers/text views for everything, except these damn ranking details. They've done my head in.
No matter what I tried, and despite every box looking perfectly aligned, I never managed to align the texts properly :/ with the spannables, there's no issue, but indeed it's hacky

Reply:
Fixed in https://github.com/acristescu/OnlineGo/commit/1c57f6a852393483b9a7b3936e3048083c23aed1